demos/http/http_demo_plaintext/http_demo_plaintext.c

```diff
@@ -0,0 +1,552 @@
+                        SOL_SOCKET,
+                        SO_SNDTIMEO,
+                        ( char * ) &transportTimeout,
+                        sizeof( transportTimeout ) ) < 0 )
+        {
+            LogError( ( "Setting socket send timeout failed." ) );
+            returnStatus = EXIT_FAILURE;
+        }
+    }
+
+    if( pListHead != NULL )
+    {
+        freeaddrinfo( pListHead );
+    }
+
+    return returnStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+static int32_t transportSend( HTTPNetworkContext_t * pContext,
+                              const void * pBuffer,
+                              size_t bytesToSend )
+{
+    int32_t bytesSend = 0;
+
+    bytesSend = send( ( int ) pContext->tcpSocket, pBuffer, bytesToSend, 0 );
+
+    if( bytesSend < 0 )
+    {
+        /* Check if it was time out */
+        if( ( errno == EAGAIN ) || ( errno == EWOULDBLOCK ) )
+        {
+            /* Set return value to 0 to indicate that send was timed out. */
+            bytesSend = 0;
+        }
+    }
+
+    return bytesSend;
+}
+
+/*-----------------------------------------------------------*/
+
+static int32_t transportRecv( HTTPNetworkContext_t * pContext,
+                              void * pBuffer,
+                              size_t bytesToRecv )
+{
+    int32_t bytesReceived = 0;
+
+    bytesReceived = ( int32_t ) recv( ( int ) pContext->tcpSocket, pBuffer, bytesToRecv, 0 );
+
+    if( bytesReceived == 0 )
+    {
+        /* Server closed the connection, treat it as an error. */
+        bytesReceived = -1;
+    }
+    else if( bytesReceived < 0 )
+    {
+        /* Check if it was time out */
+        if( ( errno == EAGAIN ) || ( errno == EWOULDBLOCK ) )
+        {
+            /* Set return value to 0 to indicate nothing to receive */
+            bytesReceived = 0;
+        }
+    }
+    else
+    {
+        /* EMPTY else */
+    }
+
+    return bytesReceived;
+}
+
+/*-----------------------------------------------------------*/
+
+static int sendHttpRequest( HTTPTransportInterface_t * pTransportInterface,
+                            const char * pHost,
+                            const char * pMethod,
+                            const char * pPath )
+{
+    HTTPStatus_t httpStatus = HTTP_SUCCESS;
+    HTTPRequestHeaders_t requestHeaders = { 0 };
+    HTTPRequestInfo_t requestInfo = { 0 };
+    HTTPResponse_t response = { 0 };
+
+    assert( pTransportInterface != NULL );
+    assert( pMethod != NULL );
+
+    /* Initialize the request object. */
+    requestInfo.pHost = pHost;
+    requestInfo.hostLen = strlen( pHost );
+    requestInfo.method = pMethod;
+    requestInfo.methodLen = strlen( pMethod );
+    requestInfo.pPath = pPath;
+    requestInfo.pathLen = strlen( pPath );
+
+    /* Set "Connection" HTTP header to "keep-alive" so that multiple requests
+     * can be sent over the same established TCP connection. */
+    requestInfo.flags = HTTP_REQUEST_KEEP_ALIVE_FLAG;
```

Comment:
This was fixed right? I suppose it is good to be explicit.

Reply:
It is fixed but yah, it's good to be explicit.